flutter-for-android.md

```diff
@@ -0,0 +1,1845 @@
+information in the object.
+
+For example, in Android, this would be similar to placing a TextView with 
+static text, or a logo, that isn't changing based on user interaction or 
+network data.
+
+If you want to dynamically change the UI based on network data or user 
+interaction then you have to work with StatefulWidget and tell the Flutter 
+framework that the widget’s State has been updated so it can update that widget.
+
+The important thing to note here is at the core both Stateless and Stateful 
+widgets behave the same. They rebuild every frame, the difference is the 
+StatefulWidget has a State object which stores state data across frames and 
+restores it.
+
+If you are in doubt, then always remember this rule: If a widget changes—the 
+user interacts with it, for example—it’s stateful.
+
+Let's take a look at how you would use a StatelessWidget. A common 
+StatelessWidget is a Text widget. If you look at the implemtnation of the Text 
+Widget you'll find it subclasses a StatelessWidget
+
+<!-- skip -->
+{% prettify dart %}
+new Text(
+  'I like Flutter!',
+  style: new TextStyle(fontWeight: FontWeight.bold),
+);
+{% endprettify %}
+
+As you can see, the Text Widget has no state information associated with it, it 
+renders what is passed in it's constructors and nothing more. 
+
+But, what if you want to make "I Like Flutter" change dynamically, for 
+example from clicking a FloatingActionButton?
+
+This can be acheived by wrapping the Text widget in a StatefulWidget and 
+updating it when the button is clicked.
+
+For example:
+
+<!-- skip -->
+{% prettify dart %}
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(new SampleApp());
+}
+
+class SampleApp extends StatelessWidget {
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Sample App',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new SampleAppPage(),
+    );
+  }
+}
+
+class SampleAppPage extends StatefulWidget {
+  SampleAppPage({Key key}) : super(key: key);
+
+  @override
+  _SampleAppPageState createState() => new _SampleAppPageState();
+}
+
+class _SampleAppPageState extends State<SampleAppPage> {
+  // Default placeholder text
+  String textToShow = "I Like Flutter";
+  void _updateText() {
+    setState(() {
+      // update the text
+      textToShow = "Flutter is Awesome!";
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text("Sample App"),
+      ),
+      body: new Center(child: new Text(textToShow)),
+      floatingActionButton: new FloatingActionButton(
+        onPressed: _updateText,
+        tooltip: 'Update Text',
+        child: new Icon(Icons.update),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+
+**How do I layout my Widgets? Where is my XML layout file?**
```

Comment:
can you link to this collection of layout widgets? https://flutter.io/widgets/layout/

Reply:
done